platform-dependent-components/problem-solver/cxx/sections-module/agent/add_section_agent.cpp

```diff
@@ -18,91 +16,91 @@
 
 #include "add_section_agent.hpp"
 
-using namespace scAgentsCommon;
 using namespace utils;
 
 namespace sectionsModule
 {
-SC_AGENT_IMPLEMENTATION(AddSectionAgent)
+ScResult AddSectionAgent::DoProgram(ScActionInitiatedEvent const & event, ScAction & action)
 {
-  ScAddr const & questionNode = m_memoryCtx.GetEdgeTarget(edgeAddr);
+  ScAddr const & questionNode = m_context.GetEdgeTarget(event.GetArc());
   if (!CheckActionClass(questionNode))
-    return SC_RESULT_OK;
+    return action.FinishSuccessfully();
 
   SC_LOG_DEBUG("AddSectionAgent started");
 
-  ScAddr sectionNameAddr =
-      IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, scAgentsCommon::CoreKeynodes::rrel_1);
-  ScAddr parentSectionAddr =
-      IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, scAgentsCommon::CoreKeynodes::rrel_2);
-  ScAddr langAddr =
-      IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, scAgentsCommon::CoreKeynodes::rrel_3);
+  ScAddr sectionNameAddr = IteratorUtils::getAnyByOutRelation(&m_context, questionNode, ScKeynodes::rrel_1);
```

Comment:
same comments as for previous agent + you can get all arguments with action.GetArguments<3>();